src/definition_storage/xml.php

```diff
@@ -592,19 +592,24 @@ public static function xmlToVariable( DOMElement $element )
         switch ( $element->tagName )
         {
             case 'array': {
-                $variable = array();
+                $variable = [];
 
-                foreach ( $element->getElementsByTagName( 'element' ) as $element )
-                {
-                    $value = self::xmlToVariable( ezcWorkflowUtil::getChildNode( $element ) );
+                $childLength = $element->childNodes->length;
 
-                    if ( $element->hasAttribute( 'key' ) )
-                    {
-                        $variable[ (string)$element->getAttribute( 'key' ) ] = $value;
-                    }
-                    else
-                    {
-                        $variable[] = $value;
+                for($childNo = 0; $childNo < $childLength; $childNo++) {
+                    $child = $element->childNodes->item($childNo);
+
+                    if ($child instanceof DOMElement) {
```

Comment:
We want to be sure that node is "element" type:
if ($child instanceof DOMElement && $child->tagName == 'element') {